mypy/newsemanal/semanal.py

```diff
@@ -3516,6 +3517,7 @@ def analyze_type_application(self, expr: IndexExpr) -> None:
         base = expr.base
         expr.analyzed = TypeApplication(base, types)
         expr.analyzed.line = expr.line
+        expr.analyzed.column = expr.column
```

Comment:
There are couple more similar places: `process_typevar_declaration()` and `process_newtype_declaration()` not sure if they are important.